pennylane/templates/subroutines/grover.py

```diff
@@ -121,9 +124,26 @@ def expand(self):
                 wires=self.wires[-1],
                 work_wires=self.work_wires,
             )
+
             PauliZ(self.wires[-1])
 
             for wire in self.wires[:-1]:
                 Hadamard(wire)
 
         return tape
+
+    def _matrix(self, *params):
```

Comment:
Here, maybe it's best to have this as a classmethod and to keep the signature at `    def _matrix(cls, *params):` or [`    def _matrix(cls, num_wires):`](https://github.com/PennyLaneAI/pennylane/blob/742f2ff067d969a42e380968f36456c995dee2c9/pennylane/ops/qubit/non_parametric_ops.py#L1048). The number of wires could then be passed by r[edefining the `matrix` method, see how it's done for QFT here](https://github.com/PennyLaneAI/pennylane/blob/742f2ff067d969a42e380968f36456c995dee2c9/pennylane/ops/qubit/non_parametric_ops.py#L1039).

Reply:
Because it is a classmethod everywhere else? I don't quite understand what's going on in the example you linked, could you explain please?

Reply:
The [original definition of the `matrix` property](https://github.com/PennyLaneAI/pennylane/blob/742f2ff067d969a42e380968f36456c995dee2c9/pennylane/operation.py#L326) in the `Operator` class passes the `parameters` attribute to `_matrix`:
```python
    @property
    def matrix(self):
        return self._matrix(*self.parameters)
```
This definition is inherited by every operation, including `GroverOperator`.

In the case of the `GroverOperator`, however, the matrix of the operation only depends on the number of wires. Therefore, we can define `_matrix` to take only the number of wires and no other parameters. If we do that, then we need to adjust our `matrix` definition too, such that instead of the parameters (`self.parameters`), the number of wires are passed when calling `_matrix`.

An example for this is how it's done for QFT at the places of the code linked above.

Reply:
And overall yes, we'd like to keep `matrix` as a property and `_matrix` as a classmethod. That's why we have this workaround for QFT.

Reply:
Ah, ok! Now I understand, before I didn't know that `matrix` was inherited. Thanks a lot for the explanation.